todo/src/components/reducers/index.js

```diff
@@ -0,0 +1,33 @@
+import { ADD_TODO, TOGGLE_TODO } from "../actions";
+
+const initialState = {
+  todos: [
+      {
+      todo: 'Walk the dog',
+      completed: false
+      }
+  ]
+};
+
+const rootReducer  = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_TODO:
+    const newTodo = { todo: action.payload, completed: false}
+     //if the passed in action has type "ADD_TODO" execute this code
+    return{
+      ...state, todos:[...state.todos, newTodo]
+    }
+
+
+    case TOGGLE_TODO:
+    return {
+      ...state, todos: state.todos.map(todo => action.payload === todo.todo ? {...todo, completed: !todo.completed} : todo)
+         }
```

Comment:
This is a very cool way of handling this!

Another way could be to get a copy of state, and then only change the completed status of the todo that should be changed, then return that copy. Let me know if you have questions on that!